### PR TITLE
update PT 1.10 images to remove repo.anaconda.com

### PR DESCRIPTION
**Pull request by kevinyang8**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description

PT 1.10 Anaconda Removal: E3, SageMaker, Neuron, and Graviton
Notes: created tickets documenting Neuron SageMaker test problems, and Graviton E3 tests hanging. These tests will fail, but are unrelated to the changes made and should not be blockers for the anaconda removal effort.

Commit ID: [0cea046](https://github.com/aws/deep-learning-containers/pull/1840/commits/0cea0460d3b6d223d6389727a1731961f80063d8) has CI passing for E3 and SageMaker.

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/inference/docker/1.10/py3/Dockerfile.graviton.e3.cpu [hunk 1]
```diff
@@ -45,22 +46,25 @@ RUN keytool -importkeystore -srckeystore /etc/ssl/certs/java/cacerts -destkeysto
     mv /etc/ssl/certs/java/cacerts.jks /etc/ssl/certs/java/cacerts; \
     /var/lib/dpkg/info/ca-certificates-java.postinst configure;
 
-RUN curl -L -o ~/miniconda.sh https://repo.anaconda.com/miniconda/Miniconda3-py38_4.10.3-Linux-aarch64.sh \
- && chmod +x ~/miniconda.sh \
- && ~/miniconda.sh -b -p /opt/conda \
- && rm ~/miniconda.sh
+RUN curl -L -o ~/mambaforge.sh https://github.com/conda-forge/miniforge/releases/download/${MAMBA_VERSION}/Mambaforge-${MAMBA_VERSION}-Linux-aarch64.sh \
```
Comment (arjkesh) [inline pytorch/inference/docker/1.10/py3/Dockerfile.graviton.e3.cpu:49]: not sure if there is a 1:1 mamba veresion to conda version mapping - cc @tejaschumbalkar - is this an OK to increase this version?

File: pytorch/inference/docker/1.10/py3/sdk1.17.1/Dockerfile.neuron [hunk 2]
```diff
@@ -106,7 +106,7 @@ RUN pip install --no-cache-dir -U \
     boto3
 
 RUN pip install neuron-cc[tensorflow] --extra-index-url https://pip.repos.neuron.amazonaws.com \
- && pip install "torch-neuron>=1.10.1,<1.10.2" --extra-index-url https://pip.repos.neuron.amazonaws.com \
+ && pip install "torch-neuron==1.10.1.2.1.7.0" --extra-index-url https://pip.repos.neuron.amazonaws.com \
```
Comment (arjkesh) [inline pytorch/inference/docker/1.10/py3/sdk1.17.1/Dockerfile.neuron:109]: why are we pinning specific torch-neuron here? apologies if i missed something in previous correspondence

File: pytorch/training/docker/1.10/py3/cu113/Dockerfile.sagemaker.gpu [hunk 3]
```diff
@@ -9,9 +9,9 @@ LABEL dlc_major_version="1"
 ARG PYTHON=python3
 ARG PYTHON_VERSION=3.8.10
 ARG PYTHON_SHORT_VERSION=3.8
-ARG CONDA_PREFIX=/opt/conda
 ARG METIS=metis-5.1.0
 ARG RMM_VERSION=0.15.0
+ARG MAMBA_VERSION=4.12.0-0
```
Comment (arjkesh) [inline pytorch/training/docker/1.10/py3/cu113/Dockerfile.sagemaker.gpu:14]: nit: is this arg used in this file?

Comment (kevinyang8) [reply pytorch/training/docker/1.10/py3/cu113/Dockerfile.sagemaker.gpu:14]: looks like it wasn't, good catch, I will remove

File: pytorch/inference/docker/1.10/py3/sdk1.17.1/Dockerfile.neuron (hunk 2 above)
Comment (kevinyang8) [reply pytorch/inference/docker/1.10/py3/sdk1.17.1/Dockerfile.neuron:109]: I was debugging a different Neuron failure and @aws-vrnatham explained that this is needed since Neuron SDK 1.19.0 was released which meant that not pinning the version would use an updated version which failed our CI.

File: pytorch/inference/docker/1.10/py3/Dockerfile.graviton.e3.cpu (hunk 1 above)
Comment (kevinyang8) [reply pytorch/inference/docker/1.10/py3/Dockerfile.graviton.e3.cpu:49]: There is no mamba to conda version mapping. Mamba just uses conda under the hood, but has faster package resolution. We actually end up still use the `conda` executable for installation in our Dockerfiles since there is a problem using `mamba` due to some way we configure our images. Miniforge by default installs from conda-forge and includes the `mamba` executable as well as `conda`. Miniconda (what was being used before) installs from repo.anaconda and only has `conda`.

**Comment by arjkesh** [comment]
outside of the comment about mamba version, LGTM - the merge strategy will be as follows

- revert buildspec changes
- revert toml changes
- merge
- rebase the PT 1.10 branch with these changes, ensure only changes on top are buildspec changes

File: pytorch/inference/docker/1.10/py3/Dockerfile.graviton.e3.cpu (hunk 1 above)
Comment (tejaschumbalkar) [reply pytorch/inference/docker/1.10/py3/Dockerfile.graviton.e3.cpu:49]: Should be fine to use the latest version of mamba
